plc4j/protocols/s7/src/test/java/org/apache/plc4x/java/s7/connection/S7PlcConnectionIT.java

```diff
@@ -58,7 +58,12 @@ public void setUp() {
     @After
     public void tearDown() {
         if(s7PlcConnection.isConnected()) {
-            s7PlcConnection.close();
+            try {
+                s7PlcConnection.close();
+            } catch (NullPointerException e) {
```

Comment:
We should investigate this in the future and try to find out why the NPE is thrown and fix things, but I don't have any objections to this for now.

Reply:
My bet is that its this line here: channel.eventLoop().parent().shutdownGracefully();
Perhaps the embedded channels Embedded Event Loop has no parent?

Reply:
maybe we should wrap all shutdown/close calls with something like that: https://commons.apache.org/proper/commons-io/javadocs/api-2.5/org/apache/commons/io/IOUtils.html#closeQuietly(java.io.Closeable...).
Do you still have the original stacktrace?

Reply:
The Stacktrace came from travis CI here https://travis-ci.org/apache/incubator-plc4x/builds/411177329

Reply:
route cause of failing test is `io.netty.util.concurrent.BlockingOperationException` and only NPE gets caught so the test still fails

Reply:
I added a fix for that.
Before catching the NPE an NPE was raised (trace from previous build:
java.lang.NullPointerException
	at org.apache.plc4x.java.s7.connection.S7PlcConnectionIT.tearDown(S7PlcConnectionIT.java:61))
Now as the NPE is catched a BlockingOperationException is thrown... weird.

Reply:
this is thrown on master as well, I justed wrapped the code like this:
```
 sendDisconnectRequestFuture.addListener((ChannelFutureListener) future -> {
                // Close the session itself.
                try {
                    channel.closeFuture().await();
                    channel.eventLoop().parent().shutdownGracefully();
                } catch (Exception e) {
                    e.printStackTrace();
                }

            });
```

Reply:
Yes, this is the code i refactored. Reason for the BlockingException is, that Netty does not allow the "await" call in a lambda which is excuted on its thread pool.
Thus, I changed program flow to avoid this.

Reply:
seems like the refactoring didn't solve this as this is still a netty thread? At least @chrisdutz is getting the very same exception on your branch right now... We should investigate why this

channel.closeFuture().await();
channel.eventLoop().parent().shutdownGracefully();
was there in the first place.

Reply:
@chrisdutz how is your setup?
For me, this removed the exception here for me with my S7.

Reply:
Hi Julian ... I just updated master ... could you check if this solves your problem?